File: app/src/main/java/io/github/sithengineer/motoqueiro/ui/cruising/CruisingContract.java
package io.github.sithengineer.motoqueiro.ui.cruising;

import rx.Observable;

public class CruisingContract {
  interface View extends io.github.sithengineer.motoqueiro.View {

    void setStopButtonEnabled();

    void setStopButtonDisabled();

    Observable<Void> stopClick();
  }

  interface Presenter extends io.github.sithengineer.motoqueiro.Presenter {
  }
}

File: app/src/main/java/io/github/sithengineer/motoqueiro/ui/cruising/CruisingContract.kt
package io.github.sithengineer.motoqueiro.ui.cruising

import io.reactivex.Observable

interface CruisingContract {
  interface View : io.github.sithengineer.motoqueiro.ui.View {

    fun setStopButtonEnabled()

    fun setStopButtonDisabled()

    fun stopClick(): Observable<Any>
  }

  interface Presenter : io.github.sithengineer.motoqueiro.ui.Presenter
}
